test/testUtils.js

```diff
@@ -11,13 +11,15 @@ import { expect } from 'chai'
  * @param {string} actor - actor name
  * @param {string} act - description of the act to be taken
  * @param {function(string) : *} factResolver - Function used to resolve facts to fall back on if no other method is available. Defaults to always false
+ * @param {Object<string, number[]|number>} cases - Object where the keys are createable facts and the values are the action index to use for that fact
  * @returns {Step}
  */
-function takeAction (actor, act, factResolver = () => false) {
+function takeAction (actor, act, factResolver = () => false, cases = {}) {
```

Comment:
Is there a reason you added the cases parameter? I think the behaviour could be obtained by a fact resolver, in fact, we even wrap it in this function. Maybe instead we could make a factory for factresolvers or something that handles this kind of thing?